UPGRADE-3.4.md

```diff
@@ -114,6 +114,40 @@ FrameworkBundle
    class has been deprecated and will be removed in 4.0. Use the
    `Symfony\Component\Translation\DependencyInjection\TranslatorPass` class instead.
 
+HttpKernel
+----------
+
+ * The `getCacheDir()` method of your kernel should return the non-empty value
+   from `Kernel::getWarmupDir()` when one exists; not doing so is deprecated.
```

Comment:
I'm not sure if I like that boilerplate. Isn't it possible to hide the switch between "real" cache directory and warmup directory?